src/Core/CoreLib/Array.cs

```diff
@@ -17,14 +17,15 @@ namespace System {
     [ScriptIgnoreNamespace]
     [ScriptImport]
     [ScriptName("Array")]
-    public sealed class Array : IEnumerable {
+    public sealed class Array : ICollection {
 
         [ScriptField]
-        public int Length {
-            get {
-                return 0;
-            }
-        }
+        [ScriptName("length")]
+        public int Length => Count;
```

Comment:
If we make these Extern we don't need to provide a body for the Get:
`public extern int Length { get; }`